include/triton/core/tritonrepoagent.h

```diff
@@ -0,0 +1,410 @@
+TRITONREPOAGENT_DECLSPEC TRITONSERVER_Error* TRITONREPOAGENT_AgentSetState(
+    TRITONREPOAGENT_Agent* agent, void* state);
+
+///
+/// The following functions can be implemented by an agent. Functions
+/// indicated as required must be implemented or the agent will fail
+/// to load.
+///
+
+/// Initialize an agent. This function is optional. This function is
+/// called once when an agent is loaded to allow the agent to
+/// initialize any state associated with the agent. An agent has a
+/// single state that is shared across all invocations of the agent.
+///
+/// \param agent The agent.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONREPOAGENT_ISPEC TRITONSERVER_Error* TRITONREPOAGENT_Initialize(
+    TRITONREPOAGENT_Agent* agent);
+
+/// Finalize for an agent. This function is optional. This function is
+/// called once, just before the agent is unloaded. All state
+/// associated with the agent should be freed and any threads created
+/// for the agent should be exited/joined before returning from this
+/// function.
+///
+/// \param agent The agent.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONREPOAGENT_ISPEC TRITONSERVER_Error* TRITONREPOAGENT_Finalize(
+    TRITONREPOAGENT_Agent* agent);
+
+/// Initialize a model associated with an agent. This function is optional.
+/// This function is called once when an agent model's lifecycle begins to allow
+/// the agent model to initialize any state associated with it. An agent model
+/// has a single state that is shared across all the lifecycle of the agent
+/// model.
+///
+/// \param agent The agent to be associated with the model.
+/// \param model The model.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONREPOAGENT_ISPEC TRITONSERVER_Error* TRITONREPOAGENT_ModelInitialize(
+    TRITONREPOAGENT_Agent* agent, TRITONREPOAGENT_AgentModel* model);
+
+/// Finalize for a model. This function is optional. This function is
+/// called once, just before the end of the agent model's lifecycle. All state
+/// associated with the agent model should be freed and any threads created
+/// for the agent model should be exited/joined before returning from this
+/// function.
+///
+/// \param agent The agent associated with the model.
+/// \param model The model.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONREPOAGENT_ISPEC TRITONSERVER_Error* TRITONREPOAGENT_ModelFinalize(
+    TRITONREPOAGENT_Agent* agent, TRITONREPOAGENT_AgentModel* model);
+
+/// Handle an action for a specified model. This function is
+/// required. Triton will not perform multiple simultaneous calls to
+/// this function for a given agent and model; however, there may be
+/// simultaneous calls for the agent for different models.
+///
+/// If the agent does not handle the action the agent should
+/// immediately return success (nullptr).
+///
+/// To modify the model's repository the agent must either acquire a mutable
```

Comment:
Change:
"Any modification to the model's repository must be made when 'action_type' is TRITONREPOAGENT_ACTION_LOAD. To modify..."